src/key-derivation.ts

```diff
@@ -1,32 +1,41 @@
-import BN from "bn.js";
-import {getHkdf} from "./crypto";
+import assert from "assert";
 import SHA256 from "bcrypto/lib/sha256";
+import BN from "bn.js";
 import {Buffer} from "buffer";
 
+import {getHkdf} from "./crypto";
+
 export function deriveMasterSK(ikm: Buffer): Buffer {
-    const okm = getHkdf(ikm, 48, Buffer.from("BLS-SIG-KEYGEN-SALT-", "utf-8"));
-    const okmBN = new BN(okm, 'hex', 'be');
-    const r = new BN("52435875175126190479447740508185965837690552500527637822603658699938581184513");
-    return Buffer.from(okmBN.mod(r).toArray("be", 32));
+  const okm = getHkdf(ikm, 48, Buffer.from("BLS-SIG-KEYGEN-SALT-", "utf-8"));
+  const okmBN = new BN(okm, "hex", "be");
+  const r = new BN("52435875175126190479447740508185965837690552500527637822603658699938581184513");
+  return Buffer.from(okmBN.mod(r).toArray("be", 32));
 }
 
-export function deriveChildSK(parentSK: Buffer, index: BN): Buffer {
-    const compressed_lamport_PK = parentSKToLamportPK(parentSK, index);
-    return deriveMasterSK(compressed_lamport_PK)
+export function deriveChildSK(parentSK: Buffer, index: number): Buffer {
+  assert(index >= 0 && index < 4294967296, "index must be 0 <= i < 2**32");
+  const compressedLamportPK = parentSKToLamportPK(parentSK, index);
+  return deriveMasterSK(compressedLamportPK);
 }
 
-function parentSKToLamportPK(parentSK: Buffer, index: BN): Buffer {
-    const salt = Buffer.from(index.toArray("be", 32));
-    const ikm = Buffer.from(parentSK);
-    const lamport0 = ikmToLamportSK(ikm, salt);
-    const notIkm = Buffer.from(ikm.map((value) => ~value));
-    const lamport1 = ikmToLamportSK(notIkm, salt);
-    const lamportSK = lamport0.concat(lamport1);
-    const lamportPK = lamportSK.map((value) => SHA256.digest(value));
-    return SHA256.digest(Buffer.concat(lamportPK));
+function parentSKToLamportPK(parentSK: Buffer, index: number): Buffer {
+  const salt = (new BN(index)).toArrayLike(Buffer, "be", 4);
```

Comment:
you could drop bn.js dep if you just mannually convert number to buffer

Reply:
We're using BN in `deriveMasterKey`. Probably need to keep it if we don't want to go the BigInt route.

Reply:
oh yeah, my bad